app/assets/javascripts/charts.js

```diff
@@ -1,264 +1 @@
-    .css({
-      top: pos.pageY - 40,
-      left: pos.pageX - 10,
-      'border-color': item.series.color,
-    })
-    .appendTo('body')
-    .zIndex(5000)
-    .show();
-}
-
-$previousPoint = null;
-function bind_hover_event(target, formater) {
-  $(target).bind('plothover', function(event, pos, item) {
-    if (item) {
-      $(target).css('cursor', 'pointer');
-      if ($previousPoint != item.datapoint) {
-        $previousPoint = item.datapoint;
-        $('#flot-tooltip').remove();
-        showTooltip(pos, item, formater);
-      }
-    } else {
-      $(target).css('cursor', 'default');
-      $('#flot-tooltip').remove();
-      $previousPoint = null;
-    }
-  });
-}
-
-function legend_selected(item) {
-  $(item)
-    .closest('td')
-    .toggleClass('disabled');
-  $(item)
-    .closest('td')
-    .next()
-    .toggleClass('disabled');
-  var target = $(item).parents('.statistics-chart');
-  var series = target.clone().data('series');
-  var legend = target.find('.legend').clone();
-  var data = [];
-  // Remove the data series.
-  target.find('.legend td.legendLabel:not(.disabled)').each(function() {
-    var key = $(this).text();
-    for (var i = 0; i < series.length; i++) {
-      if (series[i].label === key) {
-        data.push(series[i]);
-        return true;
-      }
-    }
-  });
-  flot_time_chart(target, data);
-  target.find('.legend').remove();
-  target.append(legend);
-}
-
-function ext_legend_selected(item) {
-  $(item)
-    .closest('td')
-    .toggleClass('disabled');
-  $(item)
-    .closest('td')
-    .next()
-    .toggleClass('disabled');
-  var target = $('.statistics-chart');
-  var series = target.clone().data('series');
-  var data = [];
-  // Remove the data series.
-  $('#legendContainer table td.legendLabel:not(.disabled)').each(function() {
-    var key = $(this).text();
-    for (var i = 0; i < series.length; i++) {
-      if (series[i].label === key) {
-        data.push(series[i]);
-        return true;
-      }
-    }
-  });
-
-  flot_time_chart(target, data, { show: false });
-}
-
-function updateChart(item, status) {
-  if (status == 'success')
-    $(item)
-      .find('.statistics-chart')
-      .flot_chart();
-  else $(item).text(__('Failed to load chart'));
-}
-
-$(function() {
-  $('[data-toggle="tooltip"]').tooltip();
-  refreshCharts();
-  $(document).on('click', '.reset-zoom', function() {
-    reset_zoom(this);
-  });
-  $(document).on(
-    'click',
-    '.legend .legendColorBox, .legend .legendLabel',
-    function() {
-      legend_selected(this);
-    }
-  );
-  $(document).on(
-    'click',
-    '#legendContainer .legendColorBox, .legendContainer .legendLabel',
-    function() {
-      ext_legend_selected(this);
-    }
-  );
-});
-
-$(window).on('resize', refreshCharts);
-
-function refreshCharts() {
-  $('.statistics-bar:visible').flot_bar();
-  $('.statistics-chart:visible').flot_chart();
-}
+tfm.tools.deprecate('charts.js file', 'React chart components', '3.1');
```

Comment:
Should we just remove this file completely? anyone attempting to use it would fail anyways

Reply:
Yeah, but they will get this deprecation message to see why it failed, but I guess they can find the info other ways :+1: